lib/rubocop/cop/style/fetch_env_var.rb

```diff
@@ -97,11 +97,41 @@ def allowed_var?(node)
 
         def used_as_flag?(node)
           return false if node.root?
+          return true if used_if_condition_in_body(node)
 
+          node.parent.send_type? && (node.parent.prefix_bang? || node.parent.comparison_method?)
+        end
+
+        def used_if_condition_in_body(node)
           if_node = node.ancestors.find(&:if_type?)
-          return true if if_node&.condition == node
+          if_node&.condition&.child_nodes == node.child_nodes ||
```

Comment:
```ruby
if ENV && x
  ENV[x]
end
```
The cop in this PR doesn't report that this code is offensive, while the current version of rubocop does. (I'm not sure if it's worth it to handle this edge case, though.)

Reply:
I see. I hadn't even envisioned such a case. Thanks for your review! I'll fix it.